### ISSUE TITLE
Docker: unable to apply cgroup configuration

### ISSUE THREAD

**Issue by envy2000** [wolveix/satisfactory-server#256]
**Describe the Bug**
Server won't stay up when I try to run it. Logs are as follows:
````docker:/opt/satisfactory-server# docker compose up
[+] Running 1/0
 ✔ Container satisfactory-server  Creat...                                 0.0s
Attaching to satisfactory-server
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: unable to apply cgroup configuration: cannot enter cgroupv2 "/sys/fs/cgroup/docker" with domain controllers -- it is in domain threaded mode: unknown
````
The container is successfully created though, as when I run `docker container start satisfactory-server`, I don't get a "container not found" error, but instead the above error is repeated.

**Your Runtime Command or Docker Compose File**
```services:
  satisfactory-server:
    container_name: 'satisfactory-server'
    hostname: 'satisfactory-server'
    image: 'wolveix/satisfactory-server:latest'
    ports:
      - '7777:7777/udp'
      - '15000:15000/udp'
      - '15777:15777/udp'
    volumes:
      - '/opt/satisfactory-server/config:/config'
    environment:
      - MAXPLAYERS=4
      - PGID=1000
      - PUID=1000
      - ROOTLESS=false
      - STEAMBETA=false
    restart: unless-stopped
    deploy:
      resources:
        limits:
          memory: 6G
        reservations:
          memory: 4G
```

**System Specs (please complete the following information):**
```
===== START ISSUE REPORT =====
OS:  Linux docker 6.6.7-0-lts #1-Alpine SMP PREEMPT_DYNAMIC Thu, 14 Dec 2023 08:49:17 +0000 x86_64 Linux
CPU: QEMU x86-64-v2-AES
RAM: 21GB/24GB*
HDD: 33GB/93GB (38% used)
===== END ISSUE REPORT =====
```

**Additional Context**
*The memory usage that the command got seems to be inaccurate, because according to proxmox (my Virtual Machine manager), the virtual machine is only using 7.5 of 24 Gb, not 21.

**Comment by wolveix** [comment]
Hey, this looks unrelated to the image. Are you able to run other containers? Have you tried googling the specific issue raised?

It's worth noting thst you're currently using a generic CPU model through your hypervisor (given QEMU being there). Per #248 and a number of other issues, this prevents things from working. I don't believe this is your issue, but if/when your underlying issue is addressed, this would _then_ be an issue. You should alter your VM's settings to change the CPU model to something non-generic.

**Comment by envy2000** [comment]
Though I can't look at the VM settings right (currently out of the house), I can confirm that other containers (such as crafty controller, ofsm, grafana, mySQL) worked fine.

**Comment by envy2000** [comment]
I'll try changing the vm settings later. The vm needs a reboot anyways, cause it currently has this weird, unrelated issue where, over time, its internal date will get off (if I run `date` the time is about 2 min 12 sec behind), which isn't a *huge* issue, but is still an issue.

**Comment by wolveix** [comment]
@envy2000 did you get a chance to try this?

**Comment by envy2000** [comment]
I did get a chance, but I've only seen 2 processor options work thus far: QEMU x-86-64-v2-AES (the one it's been using in the past), and QEMU x-86-64-v3 (what I have it running with now). Notably, the vm did *not* work with QEMU x-86-64-***v4***. I tried a few AMD and Intel options (I don't remember which ones), but the vm failed to boot for every one of them. For the time being, I decided I would run the satisfactory dedicated server on my Windows server (another vm in my system), which i usually use for minecraft servers that I am unable to run on my crafty controller. After doing some meddling in order to allocate more disk space, It seems to be working just fine on the Windows server.

**Comment by wolveix** [comment]
Yeah, it's an unrelated issue to the Docker image. You'd just need to change the model that the VM OS sees to something real, rather than QEMU etc.

Anyway, glad to hear you found a working solution!

**Comment by envy2000** [comment]
***However***, before I went to a windows setup, I did try 2 other docker images, as well as through commandline on a linux-mint vm running through VMware on my *desktop*, unlike my other vms, which are running through proxmox on my server, and I had the same issue on all 3 attempts. I could connect to the server through satisfactory (if I were unable to connect, satisfactory would tell me the server was currently offline, but it didn't. Also, I could see the server logging my connection every time), but every time I connected to any these servers, instead of seeing the normal server menus, i would see the message "unable to connect to the server *API*". This leads me to think that the Linux version of the Satisfactory dedicated server might be broken right now. Just in case you find that information to be valuable (though it is slightly less relevant here)

TL;DR: tried other satisfactory server implementations for linux, both on and off docker. All failed to make a connection between client and server api. Windows worked fine, making it seem the issue is with the linux satisfactory dedicated server.

TLDR is TL;DR: linux satisfactory dedicated server *might* be broken right now

**Comment by wolveix** [comment]
No, that's an unrelated issue, being #260. Those connectivity issues have already been patched in the Docker image

**Comment by envy2000** [comment]
Huh. I guess that would explain my issue for the docker images that were otherwise working completely fine. I don't see how that would have caused issues for my no-docker attempt, though, seeing as the port didn't need to be forwarded there. As soon as I started the server, I could connect to localhost:7777 through satisfactory on my desktop, and it successfully connected to the linux mint vm and the server on said vm. 

I probably won't dig too much more into that problem here though, seeing as it is pretty much out of the scope of this issue thread, and definitely out of scope of your image (which I'm sure works great for most people. It just doesnt work for me, unfortunately).

**Comment by wolveix** [comment]
If it's an IPv4-only baremetal environment, or requires port forwarding to access, that's why :)

**Comment by wolveix** [comment]
Yeah, the image works without issues :) Glad you found a working solution for your setup, though!
